src/lib/entities/entry.ts

```diff
@@ -40,6 +43,18 @@ class Entry {
     this._fields[id] = field
   }
 
+  replaceArrayLinkForLocale (id: string, locale: string, index: number, linkId: string) {
+    const link = { sys: { id: linkId, type: 'Link', linkType: 'Entry'}}
+    const field = this._fields[id] || {}
+    const fieldArray = field[locale] as Array<any> || new Array<any>()
+
+    if (fieldArray.length < index + 1) {
```

Comment:
I don't even remember, why we do not simply assign the index. Maybe it was a case of simple thinking to complex. Do you have any concern in assigning it directly?

Reply:
I'm thinking maybe it could happen that this method isn't called for every index so you'd end up with holes in the array that then mess with things later on?

If that's not the case then I don't see a problem with just assigning it.

Reply:
But even in this case, I don't see that we would create holes. The splice replaces the item at the given index and we only call the method for indices that already exist (hence replace). @GordonApplepie Any reason you choose splice in this case instead of array assignment?